### ISSUE TITLE
Autocomplete, Add an option to not auto select the first item in the list

### ISSUE THREAD

**Issue by WayneHiller** [Megabit/Blazorise#3813]
When using the Autocomplete where FreeTyping = false it is impossible to enter only part of an items text. For example I enter Cat and the list contains Caterpillar. There is no way I can keep the Cat text I entered it will auto change to Caterpillar.  If the top item was not auto selected I could keep Cat.

Are there any current workarounds to this that I might be missing?

**Comment by stsrki** [comment]
@David-Moreira can you help?

**Comment by David-Moreira** [comment]
Can you please provide an example? 
If multiple = false, seems to work fine.

See our demo for Autocomplete : https://bootstrapdemo.blazorise.com/tests/components

However if **Multiple + FreeTyping**, then it does seem to have a behaviour like you said, if the dropdown is shown with options you cannot commit the text you've entered. Is this what you're reporting? 
That is because 'Enter' key will commit the selected option. Now, I'm not sure that is wrong per say, but I understand the need to commit values that are part of the text of an option. 

Maybe if FreeTyping is activated, then it would not pre-select the first option? @stsrki what do you think?

**Comment by WayneHiller** [comment]
Sorry I had originally stated FreeTyping = false but it was supposed to be FreeTyping = true. Multiple = false. The problem as you can see in the code is that the FilterData method always sets the current Index to 0. So there is no way to enter and keep text when any of the items match the filter. The only way I found to do it is to put spaces on the end of the text until it no longer matches any item but that is funky and users will never remember that.

Probably the best would be to not select any item on Filter when FreeTyping = true.

**Comment by David-Moreira** [comment]
Right. That seems to be a good solution.
Can you however confirm that this statement however works? 
> But if you leave the Autocomplete, the text you've inputted remains.

**Comment by WayneHiller** [comment]
@David-Moreira I don't think it can work because the selected item will automatically be applied as soon as the edit looses focus. At least it does when tabbing out of it

**Comment by David-Moreira** [comment]
You just have to click outside. Eitherway, that's not a very sound workaround.

**Comment by stsrki** [comment]
Well, as far as I understand it, free-typing is meant to work only for items that are not matched at all. So it behaves as expected if you search for something, and the search result finds something partial. and then when you click on ENTER to select it.

I guess, we could add a parameter to make it behave differently, but I'm not exactly sure if that is the way we want it to.

**Comment by WayneHiller** [comment]
Ya it is a tricky issue. But it seems that if you have FreeTyping = true you should be able to enter anything even if it partially matches items in the list. Probably the easiest would be to add a Param like DisableAutoSelection. Then it could be turned off if needed.

**Comment by David-Moreira** [comment]
@stsrki I have to agree and can't hurt to implement `AutoSelection` parameter, What do you think?

**Comment by stsrki** [comment]
I agree. But it might be an opt-in rule.
